### ISSUE TITLE
Fluxion stucks at "Starting Captive Portal access point service" on Raspberry Pi 3 after a while

### ISSUE THREAD

**Issue by Xottabich95** [FluxionNetwork/fluxion#810]
Hi,
I am running Fluxion V5.9 on my Raspberry Pi 3 with Kali-Linux.
Everything works fine, but after a period of time (after a few hours or after 1-2 days), Fluxion stucks at "Starting Captive Portal access point service".

I use hostapd as Rogue AP.
But if I use airbase-ng instead of hostapd, Fluxion doesn't even start and stucks also at 
"Starting Captive Portal access point service".

I tried with Kali Linux version 2019.3 - 4.19.66-Re4son-v7+ and version 2019.4 - 4.19.81-Re4son-v7+. 
I also tested it with Ubuntu Mate - 4.15.0-1032-raspi2, and there I get the same problem.

I use two WiFi devices, one for target tracking and the other one for jamming and the access point.

#### FLUXION Info
FLUXION V5.9

#### BASH Info 
GNU bash, version 5.0.11(1)-release (arm-unknown-linux-gnueabihf)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
**Path:** /usr/bin/bash

#### Interface (wlan0) Info 
**Device**: phy0
**Driver:** rt2800usb
**Chipset:** Ralink Technology, Corp. RT2870/RT3070
**Master Modes** Yes
**Injection Test:** Injection is working!

#### XTerm Infos
**Version:** XTerm(351)
**Path:** /usr/bin/xterm
Test: XServer/XTerm success!

#### HostAPD Info
hostapd v2.9
User space daemon for IEEE 802.11 AP management,
IEEE 802.1X/WPA/WPA2/EAP/RADIUS Authenticator
Copyright (c) 2002-2019, Jouni Malinen <j@w1.fi> and contributors
Path: /usr/sbin/hostapd

#### Aircrack-ng Info
EXIT
  Aircrack-ng 1.5.2  - (C) 2006-2018 Thomas d'Otreppe
  https://www.aircrack-ng.org

#### Pyrit Info
Pyrit 0.5.1 (C) 2008-2011 Lukas Lueg - 2015 John Mora
https://github.com/JPaulMora/Pyrit
This code is distributed under the GNU General Public License v3+
**Scapy Version:** 2.4.3
Warning: Pyrit has been reported to be incompatible with scapy version 2.4.0 and above. Consult the wiki for further information. This should not affect you, if you don't choose to use pyrit in the script.                                               

#### System Info
**Chipset:** Linux version 4.19.81-Re4son-v7+ (root@kali-pi-dev) (gcc version 8.2.0 (Debian 8.2.0-6)) #1 SMP Wed Nov 6 10:16:47 AEDT 2019

#### Chipset
Chipset: rt2800usb
Chipset support ap mode
Chipset support monitor mode
Don't reseve all required information

#### FLUXION Info
FLUXION V5.9

#### BASH Info 
GNU bash, version 5.0.11(1)-release (arm-unknown-linux-gnueabihf)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
**Path:** /usr/bin/bash

#### Interface (wlan2) Info 
**Device**: phy2
**Driver:** rt2800usb
**Chipset:** Ralink Technology, Corp. RT2870
**Master Modes** Yes
**Injection Test:** Injection is working!

#### XTerm Infos
**Version:** XTerm(351)
**Path:** /usr/bin/xterm
Test: XServer/XTerm success!

#### HostAPD Info
hostapd v2.9
User space daemon for IEEE 802.11 AP management,
IEEE 802.1X/WPA/WPA2/EAP/RADIUS Authenticator
Copyright (c) 2002-2019, Jouni Malinen <j@w1.fi> and contributors
Path: /usr/sbin/hostapd

#### Aircrack-ng Info
EXIT
  Aircrack-ng 1.5.2  - (C) 2006-2018 Thomas d'Otreppe
  https://www.aircrack-ng.org

#### Pyrit Info
Pyrit 0.5.1 (C) 2008-2011 Lukas Lueg - 2015 John Mora
https://github.com/JPaulMora/Pyrit
This code is distributed under the GNU General Public License v3+
**Scapy Version:** 2.4.3
Warning: Pyrit has been reported to be incompatible with scapy version 2.4.0 and above. Consult the wiki for further information. This should not affect you, if you don't choose to use pyrit in the script.                                               

#### System Info
**Chipset:** Linux version 4.19.81-Re4son-v7+ (root@kali-pi-dev) (gcc version 8.2.0 (Debian 8.2.0-6)) #1 SMP Wed Nov 6 10:16:47 AEDT 2019

#### Chipset
Chipset: rt2800usb
Chipset support ap mode
Chipset support monitor mode
Don't reseve all required information

**Comment by strasharo** [comment]
Check your dmesg once this happens and post the output here.

**Comment by azamet90** [comment]
the best way to fix the problems with fluxion is installing from usamasattar edited version type in google usamasattar fluxion you will find it and have no errors

**Comment by strasharo** [comment]
@azamet90 his fixes are already merged.

**Comment by Xottabich95** [comment]
I did some tests and found out that when I run the Raspberry with Fluxion on my AVM FritzBox 6490 or 7590, Fluxion is stucked after 24 to 36 hours. 
But if I run Fluxion on my notebook, with the same wifi device what I used on the Raspberry and run fluxion on the same AP 6490 or 7590, Fluxion will run for 2-3 weeks without problems.

Now the confusing one.
My cousin gave me a router from the company Sercomm that I could test. When I test Fluxion on this router, Fluxion stucks after exactly 2-3 minutes, no matter if on a Raspberry or on a notebook.
That completely confused me.

I will post the output of "dmesg" right away when I test it again on the Sercomm, so I don't have to wait as long as with my AVM routers.

**Comment by strasharo** [comment]
I suspect a driver crash and if that's the case, there's isn't much that can be done on the Fluxion side.

**Comment by Xottabich95** [comment]
I can also imagine that it is due to a driver crash.
But why does Fluxion stucks after 24-36 hours on my AVM AP and after 2-3 minutes on a Sercomm AP?
Please do not close the thread. I will definitely upload the dmesg output and some screenshots today or tomorrow.
I want to figure this out.

**Comment by Xottabich95** [comment]
So, it happened again, didn't take long :)
I started Fluxion again on the Sercomm AP and after exactly 6 minutes Fluxion got stuck, see screenshot.
![screen](https://user-images.githubusercontent.com/59069368/71475130-3f374b00-27df-11ea-9033-8ad9f71a0715.png)

I emptied dmesg before starting Fluxion and that's the output after Fluxion getting stuck.
`[ 1652.090805] rt2800usb 1-1.4:1.0 fluxwl0: renamed from wlan2
[ 1652.169692] ieee80211 phy3: rt2x00lib_request_firmware: Info - Loading firmware file 'rt2870.bin'
[ 1652.169722] ieee80211 phy3: rt2x00lib_request_firmware: Info - Firmware detected - version: 0.36
[ 1657.286859] device fluxwl0 entered promiscuous mode
[ 3463.142752] rt2800usb 1-1.2:1.0 fluxwl1: renamed from wlan0
[ 3463.468091] IPv6: ADDRCONF(NETDEV_UP): fluxwl1: link is not ready
[ 3463.468136] IPv6: ADDRCONF(NETDEV_CHANGE): fluxwl1: link becomes ready
[ 3537.600880] IPv6: ADDRCONF(NETDEV_UP): fluxwl0v: link is not ready
[ 3537.622032] IPv6: ADDRCONF(NETDEV_CHANGE): fluxwl0v: link becomes ready
[ 4670.004954] IPv6: ADDRCONF(NETDEV_UP): fluxwl0v: link is not ready
[ 4670.632509] IPv6: ADDRCONF(NETDEV_UP): fluxwl0v: link is not ready
[ 4681.804347] IPv6: ADDRCONF(NETDEV_UP): fluxwl0v: link is not ready
[ 4681.926567] IPv6: ADDRCONF(NETDEV_UP): fluxwl0v: link is not ready`

**Comment by strasharo** [comment]
What exactly is hung?

**Comment by Xottabich95** [comment]
When Fluxion is running 6 windows are displayed: DHCP, DNS, AP Service, Web Service, AP Authenticator and Jammer Service.
After a while the 6 windows are closed and only the one window see above in my screenshot is shown. 
So Fluxion is no longer running.

I am currently testing it with the new version 6.1 of Fluxion that I saw today. But there I found a bug in the code, the new option "Select a method of deauthentication" doesn't start a Jammer Service window when I choose mdk3, because the mdk3 option uses the mdk4_blacklist.lst, which is probably empty.
In addition, the mdk4 option still uses mdk3 in the code and the mdk3_blacklist.lst.
Is the bug already known?

**Comment by strasharo** [comment]
Caused by a bad merge of @usama7628674 fixes, retry.

**Comment by Xottabich95** [comment]
Great, just seen the Bug is marged. 
I just tried the newest version Fluxion 6 rev.2 on my AVM AP with the deauthentication method aireplay, and Fluxion also stucks after a short time.

dmesg looks good, except for the messages "[4670.004954] IPv6: ADDRCONF (NETDEV_UP): fluxwl0: link is not ready" no special messages.

I'm at my wits' end.

**Comment by strasharo** [comment]
Are you running on a physical machine or inside a VM?

**Comment by Xottabich95** [comment]
I'm testing it on a Raspberry 3 physical, but I have the same problem on a Notebook also physical.

**Comment by strasharo** [comment]
What distribution are you running on the physical one

**Comment by Xottabich95** [comment]
As already written in the first post, I use Kali 2019.4 but also tried Ubuntu Mate and had the same problem.

**Comment by strasharo** [comment]
Works for me with the same chipset (AWUS036NH) on 2019.4 on both physical and virtual. Try with another adapter.

**Comment by Xottabich95** [comment]
Just found it, the bug has been posted before. 
But was immediately closed without a solution.
#754 
I've tried four adapters I doubt it is because of it.

**Comment by jinglei207** [comment]
> Just found it, the bug has been posted before.
> But was immediately closed without a solution.
> #754
> I've tried four adapters I doubt it is because of it.

Have you solved the problem?

**Comment by Xottabich95** [comment]
No, I still haven't been able to solve the problem. But since the new version 6.4 I noticed that on the same hardware the stuck of Fluxion happens much faster when using aireplay-ng as jammer than when using mdk4 or mdk3.

I tried with two Raspberry Pi 3 and currently with four Adapters and tested it with a notebook, but always the same problem.

I just don't know what triggers the problem, if someone knows more or has tips, I'm open to any hint.

I have a guess, but I can't find the location in the code. I suspect that maybe Fluxion restarts too fast after a channel of the AP change and the new channel is simply not known and this leads to the stuck of Fluxion. I can't find the code position to see if there is a time for the restart or if Fluxion is restarted when all data is known after the AP restart. But that's just my guess.
